src/helpers/fetch-product-cards.ts

```diff
@@ -0,0 +1,15 @@
+import { getAllProducts } from '../app/api';
+import { createProductList } from '../components/catalog/product-list';
+
+export async function fetchProductCards(layout: HTMLDivElement): Promise<void> {
+  const data = await getAllProducts();
+
+  try {
+    if ('results' in data) {
+      const cards = createProductList(data);
+      layout.append(cards);
+    }
+  } catch {
+    throw new Error('Failed to fetch products');
+  }
+}
```

Comment:
You're calling getAllProducts() before the try/catch statement, so any errors it throws won't be caught. It's better to move it inside the try block.

```suggestion
  try {
   const data = await getAllProducts();
  
    if ('results' in data) {
      const cards = createProductList(data);
      layout.append(cards);
    }
  } catch {
    throw new Error('Failed to fetch products');
  }
}
```

Reply:
Good catch! Done.